### DISCUSSION TITLE
Cordless (DECT) Phones disabled on private server's router after session

### DISCUSSION THREAD

**Discussion by jhschwarz** [jamulussoftware/jamulus#1367] [Technical Support & Help]
Any help/suggestions for troubleshooting this issue would be greatly appreciated.

A few weeks ago, our a cappella chorus had its first Jamulus session with > 12 participants using a private server on my Mac, and apart from startup/learning issues, it went well. However, when I tried to use my DECT (Gigaset 260) cordless phones after the server had been shut down, I discovered that outgoing calls from any of the phones were no longer possible (always got a busy signal). Several router reboots and a reset of the Gigaset base station eventually restored telephone function, though at a loss of Gigaset phone directories, call history and Caller ID settings. 

On Thursday, our choral group had a second, **very** satisfactory > two hour practice session with 16-17 participants using the same server. Once again, all the phones in my apartment could no longer initiate calls. Conclusion: this is not a one-time fluke.

The server itself performs very well, and doesn't seem to have any problem supporting the 17 participants. I did not update the server to 3.7.0 nor encourage the rest of our group to update their clients, because getting everybody configured and running before the 3.7.0 release took a couple of weeks, and I was not eager to repeat that experience just now; I couldn't tell from the 3.7.0 release notes whether there would be compatibility issues if we had the server at 3.7.0 and most clients still at 3.6.2.
### Details
**Location**: Server and almost all clients in Milan, Italy; one client (temporarily because of lockdown) in Venice
**Server Bandwidth** (from Fast.com): 520 Mbps download, 92 Mbps upload
**Internet Provider**: Infostrada/Wind Tre -- FTTH connection
**Server and 1 Client**: MacBookAir (13", mid-2012) with 8 GByte memory, 1 TB SSD, Intel i7 @ 2 GHz running MacOS Catalina (10.16.7); PreSonus AudioBox iTwo
**Additional client on LAN**: MacBookAir (Just new enough to make the cut for Big Sur) 
**Remote clients**: 15, Mixture of Mac and Windows

**Jamulus**: v. 3.6.2 launched in background with:

`#! /bin/bash

timestamp=$(date +"%Y-%m-%d_%H-%M-%S")

/Applications/Jamulus.app/Contents/MacOS/Jamulus -s -u 20 --norecord -l ~/Jamulus/Logs/jam$timestamp.txt`

**Overall delay on LAN**: ~ 25 ms
**Router**: Alcatel_Lucent I-240W-Q GPON Home Gateway, port forwarding for UDP 22124

**Log attached**: 
[jam2021-03-25_18-04-24.txt](https://github.com/jamulussoftware/jamulus/files/6215929/jam2021-03-25_18-04-24.txt)

**Comment by gilgongo** [comment]
How are you doing the port forwarding? Is it to the LAN address of the server only, or to the whole network?

**Reply by jhschwarz** [reply]
Only the LAN address of the server.

**Reply by gilgongo** [reply]
OK just checking that in case there was something to do with incoming UDP packets that was screwing up the phone. 

This sounds unfortunately like an issue with your particular router. Perhaps it just doesn't like the UDP-ness of Jamulus in some way (which might interfere with SIP in some way perhaps?)

**Comment by ghost** [comment]
My brother just had a problem similar to this. He does not use Jamulus. I don't have intricate details at the moment, but he phoned his service-provider and reset everything. The problem may be the phones, and/or service-provider-router in his case.

**Reply by gilgongo** [reply]
Yes, this feels like something either the router manufacturer, or the SIP provider's tech support should be handling really. Not sure anything can be done from Jamulus's end since it just handles packets.

**Reply by dcorson-ticino-com** [reply]
Very strange, I think that, at least here in Europe, most people have the telephone with voice over IP as you do and this is the first time we have heard of such a problem.  My wife is often on the phone during my sessions and all works afterward as well.
I agree that this sounds like a router problem.
For what it is worth, you are welcome to use my server (strings server, genre Classical) for your choir, from Milano you should have good ping times.

**Reply by gene96817** [reply]
There are two issues here. Since I don' t have the manual for the cordless phones I will give the general strategy for solving this problem.

First, on the router, reserve the ports needed by the cordless phones. You don't want the router's DHCP assigning the ports to a different app.

Second, on your Mac, reserve the ports used by your cordless phones. This will prevent Jamulus from using these ports as new users connect.

**Reply by jhschwarz** [reply]
Thanks for the suggestion, Gene, but it doesn't look like it's a solution to my problem. The cordless phones are from the Gigaset C620 series and as far as I can tell they are analog, **_not_** VoIP. (There is a C620 IP model, but that's not what I have. )There is no IP address to be found on the phones, much less any indication of a port. The base station, a C620A is plugged into the "Tel1" slot on the router. After looking (many times) at all settings on the router I don't see any way to "reserve" ports for the phones (really one base station and four handsets; they are not independent phones). So how phone calls get carried over the internet in this case is well hidden, AFAIK.

**Reply by gene96817** [reply]
Well, this is a puzzle. if your analog telephone set is plugged into a Tel1 socket on the router, then the router is managing all the "VoIP" functions. If your phone function is disabled after a Jamulus session, the only conclusion is that there is a bug in the router. Have you been resorting to restarting the router after a session to restoring your telephone service?  (As a long shot, after a session, you could try unplugging and plugging in your telephone. Maybe this would unwedge the router code.)

**Reply by jhschwarz** [reply]
Yes, the first time this happened I powered the router off and on. That didn't work. Then I reset the base station for the telephones and reregistered all the handsets, and this worked. The second time I started with resetting the base station but this time it didn't work, so I reset the router (losing the port forwarding for Jamulus -- not a big problem) and when this didn't work I reset the base station again at least once and things finally got working again.

**Reply by gene96817** [reply]
What I see from your description is that the router VoIP code is the "dial tone" handling is broken. First reboot the router, this also resets the router's dial-tone state machine. (Now the dial tone state machine is out of sync with your telephones.) Reset the base station, now the analog port will reset and it will try to acquire a dial-tone. Acquiring a dial-tone will now put the router and base station in the same state. 

Check if your router has a firmware update. Maybe you will be lucky.

**Reply by WolfganP** [reply]
It may be that an internal table or process in the router is flooded by the traffic and that affects your VOIP service (also managed by the router).
Unless your provider has a firmware update available and that the changelog covers that issue, I suggest ask them to set the modem in bridge mode and add your own router.

**Reply by jhschwarz** [reply]
The provider will deliver a new modem/router next week, but couldn't tell me what kind.

**Comment by ghost** [comment]
@jhschwarz 
> A few weeks ago, our a cappella chorus had its first Jamulus session with > 12 participants using a private server on my Mac, and apart from startup/learning issues, it went well. However, when I tried to use my DECT (Gigaset 260) cordless phones after the server had been shut down, I discovered that outgoing calls from any of the phones were no longer possible (always got a busy signal). Several router reboots and a reset of the Gigaset base station eventually restored telephone function, though at a loss of Gigaset phone directories, call history and Caller ID settings.
> 
> On Thursday, our choral group had a second, **very** satisfactory > two hour practice session with 16-17 participants using the same server. Once again, all the phones in my apartment could no longer initiate calls. Conclusion: this is not a one-time fluke.

@DavidSavinkoff
> My brother just had a problem similar to this. He does not use Jamulus. I don't have intricate details at the moment, but he phoned his service-provider and reset everything. The problem may be the phones, and/or service-provider-router in his case.

Additionally my brother's intercom connection for entering the building entrance stopped working, and my brther had to speak to the building manager to get this working too. What happened is that someone changed some 'settings that are related to intercoms/internet/telephones in the building', so give your manager a shout too.
